### PR TITLE
Fix issues with duplicated MatrixEvent objects around threads

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/21466
Believably fixes https://github.com/vector-im/element-web/issues/21533

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

File: src/event-mapper.ts [hunk 1]
```diff
@@ -29,9 +29,22 @@ export function eventMapperFor(client: MatrixClient, options: MapperOpts): Event
     const decrypt = options.decrypt !== false;
 
     function mapper(plainOldJsObject: Partial<IEvent>) {
```
Comment (gsouquet) [inline src/event-mapper.ts:31]: I think there would be value in testing that we're not creating dupes. 
Especially given some of the nasty that can bring

**Review by gsouquet — CHANGES_REQUESTED** [review]
Otherwise looks good to me!

File: src/event-mapper.ts (hunk 1 above)
Comment (t3chguy) [reply src/event-mapper.ts:31]: Good call, will do so shortly

**Review by gsouquet — APPROVED** [review]
LGTM 👍  thank you for the tests

**Pull request merged**